posts/2022-07-12-components-with-dynamic-attributes.md

```diff
@@ -0,0 +1,272 @@
+
+  end
+
+  def render(assigns) do
+    ~H"""
+    <h1>Board</h1>
+    <h2>Weekly Board Task </h2>
+    <div class="row">
+      <div class="column">
+        <h3>Work</h3>
+        <%= for card <- @work_cards do %> 
+          <.card card={card} />
+        <% end %>
+       </div>
+       <div class="column">
+         <h3>House</h3>
+         <%= for card <- @house_cards do %> 
+           <.card card={card} />
+         <% end %>
+       </div>
+       <div class="column">
+         <h3>School</h3>
+         <%= for card <- @school_cards do %> 
+           <.card card={card} />
+         <% end %>
+        </div>
+      </div>
+    """
+end
+
+def card(assigns) do
+    ~H"""
+    <div>
+      <div class="column">
+        <%= @card.task %>
+      </div>
+    </div>
+    """
+  end
+end
+
+```
+
+I have created a BoardLive page that renders each card component for house, work and school section.
+
+The card components have been defined using [function components](https://hexdocs.pm/phoenix_live_view/0.16.0/Phoenix.Component.html) mechanism, which is useful for reusing markup in our LiveView.
+
+**NB:** I’m not getting into the details of how the user should add their weekly task. In this example I have hard-coded the task assuming that the user had already added their task. We are actually displaying the task added to the cards .
+
+Our weekly planning task board should look something similar to this when we open our browser:
+
+
+![board plan output](/images/board-plan-output.png)
+
+
+## 2.Problem
+
+  1. We want the cards in each section to have a different color from one another . For example, cards in work section can be blue, in house can be green and in school section can be yellow.
+
+  2. Card component defined earlier should also be maintained.
+
+
+Imagine writing our markup in this manner:
+
+```elixir
+    <div class="row">
+      <div class="column">
+        <h3>Work</h3>
+        <%= for card <- @work_cards do %> 
+         <div>
+          <div class="column bg-blue">
+            <%= card.task %>
+          </div>
+         </div>
+        <% end %>
+       </div>
+       <div class="column">
+         <h3>House</h3>
+         <%= for card <- @house_cards do %> 
+           <div>
+            <div class="column bg-green">
+             <%= card.task %>
+            </div>
+          </div>
+         <% end %>
+       </div>
+       <div class="column">
+         <h3>School</h3>
+         <%= for card <- @school_cards do %> 
+          <div>
+           <div class="column bg-yellow">
+            <%= card.task %>
+           </div>
+         </div>
+         <% end %>
+        </div>
+      </div>
+
+```
+
+We have solved our first problem , its working but it looks so messed up. Personally, its redundant to write the same markup over and over again and this is the reason why we opted to use function components. However, when we use reusable components we are going back to our previous implementation.
```

Comment:
"messed up" is a fairly subjective term. I'd use more objective language like: "we can do better" or "this code has some shortcomings".